cf/data/data.py

```diff
@@ -3431,13 +3431,31 @@ def _binary_operation(self, other, method):
         else:
             result = getattr(dx0, method)(dx1)
 
+        # Set axes when other has more dimensions than self
+        axes = None
+        ndim0 = dx0.ndim
+        if not ndim0:
+            axes = other._axes
+        else:
+            diff = dx1.ndim - ndim0
+            if diff > 0:
+                axes = list(self._axes)
+                for _ in range(diff):
+                    axes.insert(0, new_axis_identifier(tuple(axes)))
```

Comment:
This block is obviously very similar logically to the block with the same purpose added to `reshape`, but since there are only two cases I think, as you probably also thought when writing it up, that it's okay to have that duplication. If there are any more methods we want to add equivalent logic to, however, I do think we should create a little internal helper function to consolidate it, though.

Reply:
Agreed. With only two cases that are quite the same what have is OK. We'll see if more similar cases arise and decide whether to act at that time.